### ISSUE TITLE
Adding WiFi drivers not included in distro

### ISSUE THREAD

**Issue by Revalvier** [PJ-Singh-001/Cubic#158]
What I tried:

1. Installing a new generic kernel, dkms and rtl8823ce (1st try from apt, 2nd try from sources in github)

2. Added grub.cfg commands for net.ifnames=0 so no name changes of adapters if detected.

3. Added a persistent rule in /etc/udev/rules.d for non-changing interface names (on the host it tends to change itself unless this is done)

What happens:

1. At boot time, the custom distro (it is already customized by my company, and was asked to just buy a new one instead of adding support) will say "no wireless card detected, please press refresh after connecting to the internet". You cannot proceed to anywhere, no login via tty1 onwards. Since I don't even have access to root so dmesg is not possible either.

The device is using the rtl8823 chipset.

**Comment by PJ-Singh-001** [comment]
You should be able to install the driver while customizing your OS using Cubic.

Here are some links I found for similar problems; perhaps you have already tried these techniques?

- https://dev.to/limal/install-rtl88x2ub-drivers-on-ubuntu-20-10-1ofp

- https://ubuntuhandbook.org/index.php/2021/01/install-rtl8188eu-rtl8723ds-drivers-ppa-ubuntu-20-04/

- https://askubuntu.com/questions/1263141/rtl8822be-driver-for-ubuntu-18-04-and-20-04

I believe drivers are installed to a directory corresponding to the Linux kernel you are *currently* running. Make sure your driver is listed in the `/lib/modules/<kernel version>` directory. For example:

    /lib/modules/2.6.31-19-generic

In Cubic, the current kernel is shown at the top right of the [Terminal page](https://github.com/PJ-Singh-001/Cubic/wiki/Terminal-Page).

It is possible you are running a different kernel on your computer, than the one that the Live ISO will eventually install to your computer. As a result, the driver may have been installed for the wrong kernel.

You can see which kernels are available to your Live ISO by executing the following on the [Terminal page](https://github.com/PJ-Singh-001/Cubic/wiki/Terminal-Page). 

    ls -la /boot

The matchig kernel must also be selected on the [Kernel tab](https://github.com/PJ-Singh-001/Cubic/wiki/Options-Page#kernel-tab).



Also, don't forget to update the initramfs.

    update-initramfs -uk all

Lastly, it may be easier to simply run Cubic on your host computer using the *same* kernel version as the default/original Live ISO you are attempting to customize.

**Comment by Revalvier** [comment]
Turns out it is due to the particulars of this custom distro plus the laptop itself.

1. Bug is affecting distros using Ubuntu 18.04 as base when it comes to persistence, meaning persistence cannot be done in the same device as the booting USB. (regarding /cow overlay on root failing)
2. Lenovo Thinkpads having a weird version of the InsydeH2O BIOS for non-TPM models. (causes non-conventional udev naming schemes). Most E570s have a TPM, except models shipped from China and Lenovo gives a "modified" version of their BIOS compared to their counterparts which causes a bug that is almost as similar as https://github.com/systemd/systemd/issues/22692. Imagine me being irritated when I found out that persistence in the USB is a MUST for the /var and /dev to have mountpoints for live-usb sessions, and with the base being Xubuntu 18.04, so it can't be done!

It is solved by formatting the USB in 64kb allocation (mitigates no. 2, allows for a "normal" detection of udev that are pointing to the NVMe and all storage and mountpoints. This problem has been solved since 19.04), adding persistence (allowed only on 18.04 not using default FAT32 allocation size) then using unetbootin with the edit of "net.ifnames=0" (mitigates no. 2 on the network drivers and udev) on the grub for the buggy udev scheme.

I thought it is about the ISO creation. Turns out it is the OS base.
